### PR TITLE
feat(ui): Create Logo Component (#12)

### LINKED ISSUE #12: Create Logo component
**Acceptance criteria:**
- All its states match mockups.
- Renders an `img` element and can receive all its props instead of `src` and `alt`.
- Has an `alt` set to "Logo CodersCamp".
- Receives the `color` prop with `white` and `black` values.
- Receives the `layout` prop with `horizontal` and `square` values.

**Mockups:**
[Figma link](https://www.figma.com/file/ur0KLA9ZOtiodrAmpXrxh9/CodersCamp-app-and-website?node-id=3%3A269)

### PR DESCRIPTION
**Pull request by asiaziola**
Closes #12

### CODE REVIEW

File: packages/ui/src/components/Logo/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type LogoColor = 'white' | 'black';
+
+export interface LogoProps extends HTMLChakraProps<'img'> {
+  layout?: LayoutVariant;
+  color?: LogoColor;
+  onClick?: MouseEventHandler;
+}
+
+const getSrcFromProps = (color: LogoColor, layout: LayoutVariant): string => {
```
Comment (KonradSzwarc) [inline packages/ui/src/components/Logo/Logo.tsx:19]: I would consider using an object map instead of a function. It could look like below:
```
const logosMap = {
    white: { horizontal: whiteHorizontalLogo, square:  whiteSquareLogo },
    black: { horizontal: blackHorizontalLogo, square:  blackSquareLogo },
}
```
In such a case, you will use `logosMap[color][layout]` as the value. It would remove the need to return`'./LogoWhiteHorizontal'` as a default value.

File: packages/ui/src/components/Logo/Logo.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const Template: Story<LogoProps> = (args) => <Logo {...args} />;
+
+export const Playground = Template.bind({});
+
+Playground.args = {
+  layout: 'horizontal',
+  color: 'black',
+};
+
+export const LayoutVariants = () => (
```
Comment (KonradSzwarc) [inline packages/ui/src/components/Logo/Logo.stories.tsx:23]: You can name it just `Layouts` as you did in the case of colors stories.

File: packages/ui/src/components/Logo/Logo.tsx (hunk 1 above)
Comment (asiaziola) [reply packages/ui/src/components/Logo/Logo.tsx]: done

File: packages/ui/src/components/Logo/Logo.stories.tsx (hunk 2 above)
Comment (asiaziola) [reply packages/ui/src/components/Logo/Logo.stories.tsx]: done

File: tsconfig.eslint.json [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
-  "include": ["**/*.js", "**/*.ts", "**/*.tsx", ".eslintrc.js"]
-}
+  "include": [
+    "**/*.js",
+    "**/*.ts",
+    "**/*.tsx",
+    ".eslintrc.js",
+    "*.svg"
+  ]
+}
```
Comment (mecies) [inline tsconfig.eslint.json:10]: Also empty line

**Review by mecies — APPROVED** [review]
Looks good to me ✅ 
I left one question, please also add empty lines at the end of the files, thank you 😄

File: packages/ui/package.json [hunk 4]
```diff
@@ -21,7 +21,8 @@
     "@chakra-ui/theme-tools": "1.1.7",
     "@emotion/react": "11.4.0",
     "@emotion/styled": "11.3.0",
-    "framer-motion": "4.1.17"
+    "framer-motion": "4.1.17",
+    "ts-node": "10.0.0"
```
Comment (KonradSzwarc) [inline packages/ui/package.json:25]: Why we need ts-node here?

File: packages/ui/src/components/Logo/Logo.test.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import blackHorizontalLogo from '../../../images/LogoBlackHorizontal.svg';
```
[the remark this answers is not in the document]
Comment (KonradSzwarc) [reply packages/ui/src/components/Logo/Logo.test.tsx:4]: Paths are not set up in this workspace, but it's a great idea to set them up 😄

File: packages/ui/src/components/Logo/Logo.test.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import blackSquareLogo from '../../../images/LogoBlackSquare.svg';
+import whiteHorizontalLogo from '../../../images/LogoWhiteHorizontal.svg';
+import whiteSquareLogo from '../../../images/LogoWhiteSquare.svg';
+import { Logo } from './Logo';
+
+describe('Logo', () => {
+  it('renders default', () => {
+    render(<Logo />);
+
+    const logo = screen.getByAltText('Logo CodersCamp') as unknown as HTMLImageElement;
```
Comment (KonradSzwarc) [inline packages/ui/src/components/Logo/Logo.test.tsx:14]: This double casting doesn't seem right. What is the reasoning behind it?

File: packages/ui/src/components/Logo/Logo.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import whiteSquareLogo from '../../../images/LogoWhiteSquare.svg';
+
+type Layout = 'horizontal' | 'square';
+
+type LogoColor = 'white' | 'black';
+
+export interface LogoProps extends HTMLChakraProps<'img'> {
+  layout?: Layout;
+  color?: LogoColor;
+  onClick?: MouseEventHandler;
```
Comment (KonradSzwarc) [inline packages/ui/src/components/Logo/Logo.tsx:16]: I think this component shouldn't receive the `onClick` prop. We probably wouldn't use it right now.

File: packages/ui/src/components/Logo/Logo.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+const logosMap = {
+  white: { horizontal: whiteHorizontalLogo as unknown as string, square: whiteSquareLogo as unknown as string },
```
Comment (KonradSzwarc) [inline packages/ui/src/components/Logo/Logo.tsx:20]: `as unknown as string` doesn't seem right. Why did you need to add them?

File: tsconfig.eslint.json [hunk 9]
```diff
@@ -1,4 +1,10 @@
 {
   "extends": "./tsconfig.base.json",
-  "include": ["**/*.js", "**/*.ts", "**/*.tsx", ".eslintrc.js"]
-}
+  "include": [
+    "**/*.js",
+    "**/*.ts",
+    "**/*.tsx",
+    ".eslintrc.js",
+    "*.svg"
```
Comment (KonradSzwarc) [inline tsconfig.eslint.json:8]: Is this line needed for code to work properly?

**Review by KonradSzwarc — CHANGES_REQUESTED** [review]

File: tsconfig.eslint.json (hunk 9 above)
Comment (asiaziola) [reply tsconfig.eslint.json]: removed

File: packages/ui/src/components/Logo/Logo.tsx (hunk 8 above)
Comment (asiaziola) [reply packages/ui/src/components/Logo/Logo.tsx]: it's a typecheck thing - it says that StaticImageData cannot be cast to string

File: packages/ui/src/components/Logo/Logo.tsx (hunk 7 above)
Comment (asiaziola) [reply packages/ui/src/components/Logo/Logo.tsx]: removed for now - but i thought that clicking on the logo should redirect user to the dashboard

File: packages/ui/src/components/Logo/Logo.test.tsx (hunk 6 above)
Comment (asiaziola) [reply packages/ui/src/components/Logo/Logo.test.tsx]: fount a workaround for this file - casting removed

File: packages/ui/package.json (hunk 4 above)
Comment (asiaziola) [reply packages/ui/package.json]: removed

File: packages/ui/src/components/Logo/Logo.test.tsx (hunk 6 above)
Comment (asiaziola) [reply packages/ui/src/components/Logo/Logo.test.tsx]: fount a workaround for this file - casting removed

File: tsconfig.eslint.json (hunk 3 above)
Comment (asiaziola) [reply tsconfig.eslint.json]: done

File: packages/ui/tsconfig.json [hunk 10]
```diff
@@ -2,13 +2,19 @@
...
     "rootDir": "./src",
     "jsx": "react",
     "noEmit": true
   },
-  "include": ["src"]
-}
+  "include": [
+    "src"
+  ]
+}
```
[the remark this answers is not in the document]
Comment (asiaziola) [reply packages/ui/tsconfig.json]: done

**Review by KonradSzwarc — DISMISSED** [review]

**Review by KonradSzwarc — APPROVED** [review]

**Review by hoolek77 — APPROVED** [review]
